tests/testthat/test-lancelot-multistrain.R

```diff
@@ -3515,3 +3516,162 @@ test_that("rotate strain uses correct variables", {
+                mean(y$ihr_strain[2, , ], na.rm = TRUE))
+  expect_true(mean(y$hfr_strain[1, , ], na.rm = TRUE) >
+                mean(y$hfr_strain[2, , ], na.rm = TRUE))
+
+
+  # Perfect VE vs strain 1 infection protects only against strain 1
+  p$rel_susceptibility <- array(matrix(c(0, 1, 1, 0),
+                                       nrow = dim[1], ncol = dim[2],
+                                       byrow = TRUE), dim = dim)
+  y <- helper(p)
+  expect_true(mean(y$ifr_strain[1, , ], na.rm = TRUE) == 0 &&
+                 mean(y$ifr_strain[2, , ], na.rm = TRUE) > 0)
+  expect_true(mean(y$ihr_strain[1, , ], na.rm = TRUE) == 0 &&
+                 mean(y$ihr_strain[2, , ], na.rm = TRUE) > 0)
+  expect_true(mean(y$hfr_strain[1, , ], na.rm = TRUE) == 0 &&
+                 mean(y$hfr_strain[2, , ], na.rm = TRUE) > 0)
+
+
+  # Perfect VE vs strain 2 infection protects only against strain 2
+  p$rel_susceptibility <- array(matrix(c(1, 0, 0, 1),
+                                       nrow = dim[1], ncol = dim[2],
+                                       byrow = TRUE), dim = dim)
+  y <- helper(p)
+  expect_true(mean(y$ifr_strain[1, , ], na.rm = TRUE) > 0 &&
+                # we have small numbers for strain 2, from seeded cases
+                 mean(y$ifr_strain[2, , ], na.rm = TRUE) < 0.001)
+  expect_true(mean(y$ihr_strain[1, , ], na.rm = TRUE) > 0 &&
+                 mean(y$ihr_strain[2, , ], na.rm = TRUE) < 0.001)
+  expect_true(mean(y$hfr_strain[1, , ], na.rm = TRUE) > 0 &&
+                 mean(y$hfr_strain[2, , ], na.rm = TRUE) < 0.001)
+
+
+  # Perfect VE vs strain 1 hospitalisation only
+  p$rel_susceptibility <- array(1, dim = dim)
+  p$rel_p_hosp_if_sympt <- array(matrix(c(0, 1, 1, 0),
+                                        nrow = dim[1], ncol = dim[2],
+                                        byrow = TRUE), dim = dim)
+  y <- helper(p)
+  expect_true(mean(y$ifr_strain[1, , ], na.rm = TRUE) == 0 &&
+                mean(y$ifr_strain[2, , ], na.rm = TRUE) > 0)
+  expect_true(mean(y$ihr_strain[1, , ], na.rm = TRUE) == 0 &&
+              mean(y$ihr_strain[2, , ], na.rm = TRUE) > 0)
+  expect_true(mean(y$hfr_strain[1, , ], na.rm = TRUE) == 0 &&
+                mean(y$hfr_strain[2, , ], na.rm = TRUE) > 0)
+
+
+  # Perfect VE vs strain 2 hospitalisation only
+  p$rel_p_hosp_if_sympt <- array(matrix(c(1, 0, 0, 1),
+                                        nrow = dim[1], ncol = dim[2],
+                                        byrow = TRUE), dim = dim)
+  y <- helper(p)
+  expect_true(mean(y$ifr_strain[1, , ], na.rm = TRUE) > 0 &&
+                mean(y$ifr_strain[2, , ], na.rm = TRUE) == 0)
+  expect_true(mean(y$ihr_strain[1, , ], na.rm = TRUE) > 0 &&
+                mean(y$ihr_strain[2, , ], na.rm = TRUE) == 0)
+  expect_true(mean(y$hfr_strain[1, , ], na.rm = TRUE) > 0 &&
+                mean(y$hfr_strain[2, , ], na.rm = TRUE) == 0)
+
+
+  # Perfect VE vs strain 1 death only
+  # this is having no effect whatsoever on ifr and hfr!!
```

Comment:
@edknock this doesn't seem to be working, and I'm not sure why. Could you have a look please?